flow/src/main/java/com/sequenceiq/flow/core/stats/FlowOperationStatisticsPersister.java

```diff
@@ -0,0 +1,81 @@
+package com.sequenceiq.flow.core.stats;
+
+import java.text.DecimalFormat;
+import java.util.Date;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.EvictingQueue;
+import com.sequenceiq.cloudbreak.common.event.PayloadContext;
+import com.sequenceiq.cloudbreak.common.service.TransactionService;
+import com.sequenceiq.flow.api.model.operation.OperationType;
+import com.sequenceiq.flow.core.cache.FlowStat;
+import com.sequenceiq.flow.domain.FlowOperationStats;
+import com.sequenceiq.flow.repository.FlowOperationStatsRepository;
+
+@Service
+public class FlowOperationStatisticsPersister {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FlowOperationStatisticsPersister.class);
+
+    private static final Integer MAX_FLOW_STAT_SIZE = 10;
+
+    private static final double MILLISEC_TO_SEC_DOUBLE_DEVIDER = 1000d;
+
+    @Inject
+    private FlowOperationStatsRepository flowOperationStatsRepository;
+
+    @Inject
+    private TransactionService transactionService;
+
+    public synchronized void save(FlowStat flowStat) {
```

Comment:
This method does too much.
Please split it up to follow some of the clean code principles.

Reply:
splitted it